### PR TITLE
Update Google HTTP, API, and OAuth libs to much newer versions

### PR DESCRIPTION
**Pull request by orrc**
TL;DR: This updates some _really_ old dependencies, and will allow me to update my plugin which depends on this one.

---

This updates the Google HTTP client (from a version released in November 2015), the Google API client (from August 2018), and the OAuth2 library (from May 2019).

Updating these libraries seems like a good idea, and is the culmination of a yak-shaving adventure I went on in order to update the Google Play client library in my [plugin](https://github.com/jenkinsci/google-play-android-publisher-plugin), which depends on this one…

The latest Google Play client library now depends on google-http-client 1.38+, which is a problem since the older version gets class-loaded from this plugin instead (as far as I understand it), and I would get `NoClassDefFoundError`s, despite depending on 1.39.2 in my plugin.

However, newer versions of `google-http-client` depend on newer Guava versions (v14+, AFAICT) due to depending on classes like [`BaseEncoding`](https://guava.dev/releases/14.0/api/docs/com/google/common/io/BaseEncoding.html) and [`StandardSystemProperty`](https://guava.dev/releases/15.0/api/docs/com/google/common/base/StandardSystemProperty.html), and so the ancient v11 bundled with Jenkins is no use.

So despite the removal of Guava from this plugin in #114, I had to pull in a newer version again, and this time applied a workaround from [JENKINS-36779](https://issues.jenkins.io/browse/JENKINS-36779?focusedCommentId=403197&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-403197).

The automated tests still pass as before (i.e. two of them fail on JDK 8), and bundling a snapshot of this PR into the Google Play Android Publisher plugin allowed it to function as expected with the upgraded Google Play client library.

I also did some other small cleanups, which should supersede #118, #124, and #126.

### CODE REVIEW

**Comment by orrc** [comment]
Hey @donmccasland and @craigdbarber — are you the current maintainers? I wasn't sure 100% sure whether the POM is up-to-date, or who gets notified at Google for PRs :)

Anyway, I would appreciate it if you manage to find some time to take a look at this upgrade. Hopefully it looks reasonable!

**Review by rsandell — APPROVED** [review]
